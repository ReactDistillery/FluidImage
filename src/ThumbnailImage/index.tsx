import * as React from "react";

export interface ThumbnailImageProps {
  src: string;
  alt?: string;
  title?: string;
  hasImageLoaded?: boolean;
  hideImageOnLoad?: boolean;
  onLoad: Function;
}

const styles: React.CSSProperties = {
  thumbnailImage: {
    width: "100%",
    height: "auto",
    position: "absolute",
    zIndex: 0,
    transition: "opacity 0.6s"
  }
};

const ThumbnailImage = ({
  onLoad,
  hasImageLoaded = false,
  hideImageOnLoad = false,
  title,
  alt,
  src
}: ThumbnailImageProps) => (
  <img
    onLoad={() => onLoad()}
    src={src}
    title={title}
    style={{
      ...styles.thumbnailImage,
      opacity: Number(!hideImageOnLoad && hasImageLoaded)
    }}
    alt={alt}
    draggable={false}
  />
);

export default ThumbnailImage;
