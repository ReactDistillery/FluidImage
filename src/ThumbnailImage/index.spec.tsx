import React from "react";
import { create } from "react-test-renderer";
import ThumbnailImage from "./";

const minimalProps = {
  src: "testImage.jpeg",
  onLoad: () => {}
};

const fullProps = {
  src: "testImage.jpeg",
  alt: "some description",
  title: "some title",
  hasImageLoaded: true,
  hideImageOnLoad: false,
  onLoad: () => {}
};

describe("FluidImage - ThumbnailImage", () => {
  test("Renders an img with minimal props", () => {
    const thumbnailImage = create(
      <ThumbnailImage {...minimalProps} />
    ).toJSON();
    expect(thumbnailImage).toMatchSnapshot();
  });
  test("Renders an img with full props", () => {
    const thumbnailImage = create(<ThumbnailImage {...fullProps} />).toJSON();
    expect(thumbnailImage).toMatchSnapshot();
  });
  test("it calls onLoad when image is loaded", () => {
    const onLoad = jest.fn();
    const thumbnailImage = create(
      <ThumbnailImage {...minimalProps} onLoad={onLoad} />
    );
    /* Had a lot of trouble getting onLoad to fire in Jest + JSDOM */
    thumbnailImage.root.findByType("img").props.onLoad();
    expect(onLoad.mock.calls.length).toBe(1);
  });
});
