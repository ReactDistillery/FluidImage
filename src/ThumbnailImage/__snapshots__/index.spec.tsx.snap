// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluidImage - ThumbnailImage Renders an img with full props 1`] = `
<img
  alt="some description"
  draggable={false}
  onLoad={[Function]}
  src="testImage.jpeg"
  style={
    Object {
      "height": "auto",
      "opacity": 1,
      "position": "absolute",
      "transition": "opacity 0.6s",
      "width": "100%",
      "zIndex": 0,
    }
  }
  title="some title"
/>
`;

exports[`FluidImage - ThumbnailImage Renders an img with minimal props 1`] = `
<img
  alt={undefined}
  draggable={false}
  onLoad={[Function]}
  src="testImage.jpeg"
  style={
    Object {
      "height": "auto",
      "opacity": 0,
      "position": "absolute",
      "transition": "opacity 0.6s",
      "width": "100%",
      "zIndex": 0,
    }
  }
  title={undefined}
/>
`;
